source/common/router/header_parser_utils.cc

```diff
@@ -25,34 +26,34 @@ std::string HeaderParser::translateMetadataFormat(const std::string& header_valu
   const re2::RE2& re = getMetadataTranslatorPattern();
   ASSERT(re.ok());
   std::string new_header_value = header_value;
-  re2::StringPiece json_array, metadata_type;
-  while (re.PartialMatch(new_header_value, re, &metadata_type, &json_array)) {
-    std::string new_format;
+  re2::StringPiece matches[4];
```

Comment:
I think 3 is enough. There are 2 capturing groups plus one for entire string.

Reply:
You are correct, thanks!